client/src/app/site/motions/modules/motion-detail/components/motion-poll/motion-poll-preview/motion-poll-preview.component.ts

```diff
@@ -1,56 +1,135 @@
+            if (data.label === 'YES') {
+                this.voteYes = data.data[0];
+            }
+            if (data.label === 'NO') {
+                this.voteNo = data.data[0];
+            }
+        }
+        this.chartDataSubject.next(chartData);
+    }
+
+    public get poll(): ViewMotionPoll {
+        return this._poll;
+    }
+
+    /**
+     * The id of the dedicated motion.
+     */
+    @Input()
+    public motionId: number;
+
+    /**
+     * Emits, when the user clicks the 'edit'-button.
+     */
+    @Output()
+    public edit = new EventEmitter<void>();
 
+    /**
+     * Subject to holding the data needed for the chart.
+     */
+    public chartDataSubject: BehaviorSubject<ChartData> = new BehaviorSubject([]);
+
+    /**
+     * Number of votes for `Yes`.
+     */
+    public voteYes = 0;
+
+    /**
+     * Number of votes for `No`.
+     */
+    public voteNo = 0;
+
+    /**
+     * The motion-poll.
+     */
+    private _poll: ViewMotionPoll;
+
+    /**
+     * Constructor.
+     *
+     * @param title
+     * @param translate
+     * @param matSnackbar
+     * @param router
+     * @param motionRepo
+     */
     public constructor(
         title: Title,
         protected translate: TranslateService,
         matSnackbar: MatSnackBar,
-        private repo: MotionPollRepositoryService,
-        private promptDialog: PromptService,
-        private router: Router
+        private router: Router,
+        private motionRepo: MotionRepositoryService,
+        public pollService: MotionPollService,
+        public pollRepo: MotionPollRepositoryService
     ) {
         super(title, translate, matSnackbar);
     }
 
+    /**
+     * Navigates the user to the poll-detail-view.
+     */
     public openPoll(): void {
         this.router.navigate(['motions', 'polls', this.poll.id]);
     }
 
-    public editPoll(): void {
-        this.router.navigate(['motions', 'polls', this.poll.id], { queryParams: { edit: true } });
+    /**
+     * Gets the type of the motion-poll.
+     *
+     * @returns The verbose type as string.
+     */
+    public getMotionPollType(): string {
+        return this.poll.poll.typeVerbose;
     }
 
-    public async deletePoll(): Promise<void> {
-        const title = 'Delete poll';
-        const text = 'Do you really want to delete the selected poll?';
-
-        if (await this.promptDialog.open(title, text)) {
-            await this.repo.delete(this.poll);
-        }
+    /**
+     * Gets the current state of the motion-poll.
+     *
+     * @returns The verbose state as string.
+     */
+    public getMotionPollState(): string {
+        return this.poll.poll.stateVerbose;
     }
 
-    public enterAnalogVotes(): void {
-        throw new Error('TODO');
+    /**
+     * Searches for the parent-motion of this poll.
+     *
+     * @returns a `ViewMotion`.
+     */
+    public getMotionOfPoll(): ViewMotion {
+        return this.motionRepo.getViewModel(this.poll.motion_id);
```

Comment:
Please add a new relation in the motion poll repo and assignmenr poll repo to link polls to motions/assignments.